partiql-app/partiql-cli/src/main/kotlin/org/partiql/cli/pico/PartiQLCommand.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.cli.pico
+
+import org.partiql.cli.query.Cli
+import org.partiql.cli.shell.Shell
+import org.partiql.cli.utils.EmptyInputStream
+import org.partiql.cli.utils.UnclosableOutputStream
+import org.partiql.lang.eval.ExprValueFactory
+import picocli.CommandLine
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.InputStream
+
+@CommandLine.Command(
+    name = "partiql",
+    mixinStandardHelpOptions = true,
+    versionProvider = PartiQLVersionProvider::class,
+    descriptionHeading = "%n@|bold,underline,yellow The PartiQL CLI|@%n",
+    description = [
+        "%nThe PartiQL CLI allows query execution in two modes: Non-Interactive and Interactive (default).%n",
+        "@|bold,underline General Options|@%n",
+        "These options configure both Non-Interactive and Interactive executions.%n"
+    ],
+    showDefaultValues = true
+)
+internal class PartiQLCommand(private val valueFactory: ExprValueFactory) : Runnable {
+
+    @CommandLine.Mixin
+    internal lateinit var options: PipelineOptions
+
+    @CommandLine.ArgGroup(
+        exclusive = false,
+        heading = "%n@|bold,underline Non-Interactive (Single Query Execution)|@%n%n" +
+            "Specifying any of the below options will trigger Non-Interactive execution. " +
+            "Also, passing input through standard input will trigger its execution.%n%n"
+    )
+    internal var executionOptions: ExecutionOptions? = null
+
+    @CommandLine.ArgGroup(exclusive = false, heading = "%n@|bold,underline Interactive (Shell) Configurations|@%n%n")
+    internal var shellOptions: ShellOptions? = null
+
+    /**
+     * Run the CLI or Shell (default)
+     */
+    override fun run() {
+        when {
+            executionOptions?.query != null -> runCli(executionOptions!!, executionOptions!!.query!!.inputStream())
+            executionOptions != null && System.`in`.available() > 0 -> runCli(executionOptions!!, System.`in`)
+            System.`in`.available() > 0 -> runCli(ExecutionOptions(), System.`in`)
+            shellOptions != null -> runShell(shellOptions!!)
+            else -> runShell()
+        }
+    }
```

Comment:
nit to cut down on the assertions and density here

```suggestion
    override fun run() {
        val command = executionOptions ?: ExecutionOptions()
        val shell = shellOptions ?: ShellOptions()
        val stdin = System.`in`
        when {
            command.query != null -> runCli(command, command.query!!.inputStream())
            stdin.available() > 0 -> runCli(command, stdin)
            else -> runShell(shell)
        }
    }
```